### PR TITLE
CORE-5799 check for empty groupId and more tests

### PR DESCRIPTION
**Pull request by barrylapthorn**
Check for empty groupId during cpi upload validation

Extra tests for some validation functions.

### CODE REVIEW

File: components/chunking/chunk-db-write-impl/src/main/kotlin/net/corda/chunking/db/impl/validation/ValidationFunctions.kt [hunk 1]
```diff
@@ -28,14 +28,19 @@ import org.slf4j.Logger
  *
  * @throws ValidationException
  */
-fun getCpiFileInfo(cpiCacheDir: Path, chunkPersistence: ChunkPersistence, requestId: RequestId): FileInfo {
+fun assembleFileFromChunks(
```
Comment (barrylapthorn) [inline components/chunking/chunk-db-write-impl/src/main/kotlin/net/corda/chunking/db/impl/validation/ValidationFunctions.kt:31]: Renamed this and added tests in addition to the main jira issue checking group ids.

File: components/chunking/chunk-db-write-impl/src/main/kotlin/net/corda/chunking/db/impl/validation/ValidationFunctions.kt [hunk 2]
```diff
@@ -129,13 +134,21 @@ fun CpiPersistence.persistCpiToDatabase(
...
  * Get groupId from group policy JSON on the [Cpi] object.
  *
+ * @param getGroupIdFromJson lambda that takes a json string and returns the `groupId`
+ *
  * @throws ValidationException if there is no group policy json.
  * @throws CordaRuntimeException if there is an error parsing the group policy json.
  * @return `groupId`
  */
-fun Cpi.validateAndGetGroupId(): String {
+fun Cpi.validateAndGetGroupId(getGroupIdFromJson: (String) -> String): String {
```
Comment (barrylapthorn) [inline components/chunking/chunk-db-write-impl/src/main/kotlin/net/corda/chunking/db/impl/validation/ValidationFunctions.kt:144]: Main part of PR

File: components/chunking/chunk-db-write-impl/src/test/kotlin/net/corda/chunking/db/impl/validation/ValidateGroupIdTest.kt [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+import net.corda.libs.packaging.Cpi
+import net.corda.libs.packaging.core.CpiMetadata
+import net.corda.membership.lib.grouppolicy.GroupPolicyParser
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+import org.mockito.kotlin.doReturn
+import org.mockito.kotlin.mock
+
+class ValidateGroupIdTest {
```
Comment (barrylapthorn) [inline components/chunking/chunk-db-write-impl/src/test/kotlin/net/corda/chunking/db/impl/validation/ValidateGroupIdTest.kt:12]: Tests...

**Review by driessamyn — DISMISSED** [review]
LGTM

**Review by conal-smith — DISMISSED** [review]
LGTM

**Comment by barrylapthorn** [comment]
Of course there was a detekt warning... re-pushed.

**Review by driessamyn — APPROVED** [review]
LGTM

**Pull request merged**